onnxruntime/core/providers/webgpu/math/matmul.cc

```diff
@@ -222,35 +279,114 @@ MatMulProgram CreateMatMulProgram(const Activation& activation, std::vector<cons
                                                    ? InlinedVector<int64_t>({4, 1, 1})
                                                    : InlinedVector<int64_t>({4, 4, 1});
 
+  bool need_split_k = split_k_config.UseSplitK(is_vec4, activation.activation_kind_, batch_size, is_channels_last, dim_a_outer, dim_b_outer, dim_inner);
+
+  bool use_bias_in_matmul = has_bias;
+
+  // When Split-K is used, bias will be handled in `MatMulFillBiasBeforeSplitKProgram`
+  // instead of `MatMulProgram`.
+  if (need_split_k) {
```

Comment:
Merge all things together when need_split_k is true as possible? Currently, you have 4 `if (need_split_k)`.